jsonschema/tests/test_validators.py

```diff
@@ -1195,6 +1195,42 @@ def test_it_retrieves_unstored_refs_via_urlopen(self):
                     self.assertEqual(resolved, 12)
         urlopen.assert_called_once_with("http://bar")
 
+    def test_it_retrieves_unstored_file_refs_via_urlopen(self):
```

Comment:
I'd love to start moving towards real tests instead of the mocking nonsense that happens in the existing ones here (i.e. reading a real file that is shipped as part of the test suite).

It's very difficult to see whether these tests pass even before your change other than just needing to mock the right data into place.

Reply:
:thumbsup:
Some of the mocks are indeed quite fragile, I'll make that change as necessary. Not sure where to write to/from, perhaps a tmpfile?

Reply:
Yeah or just a real file that's in the `jsonschema.tests` directory which is likely even easier, though you still need to use like a relative file reference presumably.

Reply:
Do you have gitter or anything better for more general q's on the jsonschema codebase?

Reply:
I'm on IRC pretty often (and when I'm not I'm idling) -- there's a #json-schema channel that I'm in.

There's also the JSON Schema Slack: https://join.slack.com/t/json-schema/shared_invite/enQtMjk1NDcyNDI2NTAwLTcyYmYwMjdmMmUxNzZjYzIxNGU2YjdkNzdlOGZiNjIwNDI2M2Y3NmRkYjA4YmMwODMwYjgyOTFlNWZjZjAyNjg

Which I'm on, and you can certainly ping me on!

(And thanks again)